### PR TITLE
add support for unfurl_links + better logging

### PR DESCRIPTION
**Pull request by dbertram**
- add support for optional `unfurl_links` argument (defaults to `true`, but can be used to disable link unfurling when desired)
- use `--fail-with-body` so HTTP status of >= 400 triggers an error exit code
- use long form curl args in general for better readability/comprehensibility
- break up curl args over multiple lines to improve readability
- log raw response before processing it
- fix how this action sets its `ts` output so that it actually works!
- adds a GH workflow to this repo to enable testing changes in PRs _before_ they're merged and without having to setup private, throw-away repos that use your branch of this GH action for testing

### CODE REVIEW

File: action.yml
Comment (dbertram) [inline action.yml]: Something about how I was trying to output the `ts` output variable wasn't playing nicely with what GH was expecting:

image

I was looking more closely at their examples for how to do multi-line output and noticed I had added the equal sign in `ts=<<EOF` that wasn't in their example: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#multiline-strings

...but I'm not sure how test test this assumption before merging this PR. 😟

File: action.yml
Comment (dbertram) [inline action.yml:45]: My goals with these changes are:
- make this step fail if we get a response with HTTP status >= 400 (i.e., didn't work)
- capture the response and log it (to make debugging easier if anything _does_ go wrong)
- set the `ts` output for this action by processing the curl response

But I'm a total bash/curl newb, so please feel free to make suggestions! 🙏

File: action.yml
Comment (bradymholt) [inline action.yml:19]: Does this default to true in Slack API if not provided?

File: action.yml
Comment (bradymholt) [inline action.yml]: It might be nice to put `shell: bash` at top to get it out of the way of this long(er) script.

```
runs:
  using: 'composite'
  steps:
    - shell:bash
      run: |
         ...
```

File: action.yml
Comment (bradymholt) [reply action.yml]: You could create a workflow in a throw-away repo that references this branch like this:

```
- name: "Post Slack message"
  id: slack-post
  uses: ynab/slack-post-message-action@db/better-logging
  with:
    channel: "#my-channel"
    text: "Test 123"
- run: echo ${{ steps.slack-post.outputs.ts }}
```

**Review by bradymholt — APPROVED** [review]
Looks great.  I posted a few comments but nothing to hold this up.

File: action.yml
Comment (dbertram) [reply action.yml:19]: Their docs don't actually say what it defaults to, but my anecdotal testing seems to indicate yes

**Review by bradymholt — APPROVED** [review]

File: action.yml
Comment (dbertram) [reply action.yml]: https://github.com/ynab/slack-post-message-action/pull/3/commits/1f4882eb92b6ee6cc14e964c0f1e736d61a9bb2d

Comment (dbertram) [reply action.yml]: https://github.com/ynab/slack-post-message-action/pull/3/commits/1f4882eb92b6ee6cc14e964c0f1e736d61a9bb2d

File: action.yml
Comment (dbertram) [inline action.yml:23]: Oof. That took a lot more iteration than I care to admit, but this was the bit I was missing: to set your custom action's output you have to define where it should pull the value from. 🤦

File: action.yml
Comment (dbertram) [inline action.yml:34]: Not strictly what I set out to cover here, but it felt reasonable to validate the required inputs explicitly to (hopefully) provide a better troubleshooting experience.

File: action.yml
Comment (dbertram) [reply action.yml]: The bigger issue here was that I needed to define how the `ts` output gets its value. The `Invalid format` aspect _did_ need to be fixed, but didn't actually correct the problem I was debugging: the `ts` output not being available in later steps for actions using this shared action.

File: action.yml
Comment (bradymholt) [reply action.yml:23]: 😅

File: action.yml
Comment (bradymholt) [reply action.yml:34]: 👍

**Pull request merged**
